File: cpp/csvviewer/Interactors.cpp
#include "Interactors.h"
#include "Commandline.h"
#include "FileProvider.h"
#include "Csv.h"
#include "Record.h"
#include "Paging.h"

#include <string>
#include <vector>

class Interactors {
    void Interactors::Start(std::vector<std::string> args) {
        Commandline commandline;
        FileProvider fileProvider;
        Csv csv;
        Paging paging;

        auto filename = commandline.GetFilename(args);
        auto lines = fileProvider.ReadFile(filename);
        auto records = csv.CreateRecords(lines);
        return paging.ExtractFirstPage(records);
    }
};

File: python/csvviewer/interactors.py
from args import get_filename
from csv_reader import create_records
from file_provider import read_file
from paging import extract_first_page


def start():
    filename = get_filename()
    lines = read_file(filename)
    all_records = create_records(lines)
    records = extract_first_page(all_records)
    return records
